src/anal_ghidra.cpp

```diff
@@ -0,0 +1,3343 @@
+		return ESIL_PARM_FLOAT;
+	if(!strcmp(str, "nanF") || !strcmp(str, "infF") || !strcmp(str, "-nanF") ||
+	   !strcmp(str, "-infF"))
+		return ESIL_PARM_FLOAT;
+
+	return R_ANAL_ESIL_PARM_INVALID;
+}
+
+static long double esil_get_double(RReg *reg, RRegItem *item)
+{
+	RRegSet *regset;
+	float vf = 0.0f;
+	double vd = 0.0f;
+	int off;
+	long double ret = 0.0f;
+	if(!reg || !item)
+	{
+		return 0LL;
+	}
+	off = BITS2BYTES(item->offset);
+	regset = &reg->regset[item->arena];
+	switch(item->size)
+	{
+		case 32:
+			if(regset->arena->size - off - 1 >= 0)
+			{
+				memcpy(&vf, regset->arena->bytes + off, sizeof(float));
+				ret = vf;
+			}
+			break;
+		case 64:
+			if(regset->arena->size - off - 1 >= 0)
+			{
+				memcpy(&vd, regset->arena->bytes + off, sizeof(double));
+				ret = vd;
+			}
+			break;
+		case 80:
+		case 96:
+		case 128:
+			if(regset->arena->size - off - 1 >= 0)
+			{
+				memcpy(&ret, regset->arena->bytes + off, sizeof(long double));
+			}
+			break;
+		default: eprintf("esil_get_double: Bit size %d not supported\n", item->size); return 0.0f;
+	}
+	return ret;
+}
+
+static bool esil_set_double(RReg *reg, RRegItem *item, long double value)
+{
+	ut8 *src;
+	float vf = value;
+	double vd = value;
+
+	if(!item)
+	{
+		eprintf("esil_set_double: item is NULL\n");
+		return false;
+	}
+	switch(item->size)
+	{
+		case 32: src = (ut8 *)&vf;
+		case 64: src = (ut8 *)&vd;
+		case 80:
+		case 96:
+		case 128:
+			// FIXME: endian
+			src = (ut8 *)&value;
+			break;
+		default: eprintf("esil_set_double: Bit size %d not supported\n", item->size); return false;
+	}
+	if(reg->regset[item->arena].arena->size - BITS2BYTES(item->offset) - BITS2BYTES(item->size) >=
+	   0)
+	{
+		r_mem_copybits(reg->regset[item->arena].arena->bytes + BITS2BYTES(item->offset), src,
+		               item->size);
+		return true;
+	}
+	eprintf("esil_set_double: Cannot set %s to %lf\n", item->name, value);
+	return false;
+}
+
+static int esil_get_parm_float(RAnalEsil *esil, const char *str, long double *num)
+{
+	int ret = 0;
+	if(!str || !*str)
+		return false;
+
+	int parm_type = esil_get_parm_type_float(esil, str);
+	if(!num || !esil)
+		return false;
+
+	switch(parm_type)
+	{
+		case ESIL_PARM_FLOAT:
+			// *num = r_num_get (NULL, str);
+			sscanf(str, "%LfF", num);
+			ret = 1;
+			break;
+		case R_ANAL_ESIL_PARM_REG:
+		{
+			RRegItem *reg = r_reg_get(esil->anal->reg, str, get_reg_type(str));
+			if(reg)
+			{
+				*num = esil_get_double(esil->anal->reg, reg);
+				ret = 1;
+			}
+			break;
+		}
+		default:
+			if(esil->verbose)
+				eprintf("Invalid arg (%s)\n", str);
+
+			esil->parse_stop = 1;
+			break;
+	}
+	return ret;
+}
+
+static bool sleigh_esil_consts_pick(RAnalEsil *esil)
+{
+	if(!esil || !esil->stack)
+		return false;
+
+	char *idx = r_anal_esil_pop(esil);
+	ut64 i;
+	int ret = false;
+
+	if(R_ANAL_ESIL_PARM_REG == r_anal_esil_get_parm_type(esil, idx))
+	{
+		throw LowlevelError("sleigh_esil_consts_pick: argument is consts only");
```

Comment:
These throws in the esil handlers can be problematic since they crash the entire process if something is wrong in an esil string, which can theoretically always happen because the user can also supply esil. I think a simple print would be enough for now.